src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -62,6 +62,27 @@ def pileupSizeTask(self):
             msg = f"MSPileup pileup size task failed with error {exp}"
             self.logger.exception(msg)
 
+    def cleanupTask(self):
+        """
+        Execute cleanup task according to the following logic:
+
+        1. Fetch documents from backend database for the following conditions
+           - active=False, and
+           - empty ruleIds list; and
+           - empty currentRSEs; and
+           document has been deactivated for a while (deactivatedOn=XXX),
+        2. For those documents which are fetched make delete call to backend database
+        """
+        spec = {'active': False}
+        docs = self.mgr.getPileup(spec)
+        deleteDocs = 0
+        for doc in docs:
+            if not doc['ruleIds'] and not doc['currentRSEs']:
+                spec = {'_id': doc['_id']}
+                self.mgr.deletePileup(spec)
```

Comment:
Can you please make a new log record whenever a pileup object is deleted (saying which pileup object was deleted)?

Reply:
done